File: 0242-valid_anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        hashmap = {}
        
        if len(s) != len(t):
          return False
        
        for i in s:
          hashmap[i] = hashmap.get(i, 0) + 1
          
        for i in t:
          hashmap[i] = hashmap.get(i, 0) - 1
          
          if hashmap.get(i) < 0:
            return False
          
        return True
          
        
s = Solution()
print(s.isAnagram('ab', 'a'))

File: 0242-valid_anagram.go
func isAnagram(s string, t string) bool {
    if len(s) != len(t) {
        return false
    }
    
    count := make(map[byte]int)

    for i := range s {
        count[s[i]]++
        count[t[i]]--
    }

    for _, v := range count {
        if v != 0 {
            return false
        }
    }

    return true
} 